scheduler/workflow.go

```diff
@@ -310,17 +310,17 @@ func (s *schedulerWorkflow) Start(t *task) {
 	}).Info(fmt.Sprintf("Starting workflow for task (%s\\%s)", t.id, t.name))
 	s.state = WorkflowStarted
 	j := newCollectorJob(s.metrics, t.deadlineDuration, t.metricsManager, t.workflow.configTree, t.id)
+	if len(j.Errors()) > 0 {
```

Comment:
`j` cannot have an error here, it's just being constructed.  That being said, I think `toRecordFailure` is probably not needed, we just need the changes you made on line 322 and 323 to use `errors` instead of `j.Errors()`.